axonserver/src/main/java/io/axoniq/axonserver/message/ByteArrayMarshaller.java

```diff
@@ -0,0 +1,59 @@
+package io.axoniq.axonserver.message;
+
+import io.grpc.KnownLength;
+import io.grpc.MethodDescriptor;
+import org.apache.commons.compress.utils.IOUtils;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ByteArrayMarshaller implements MethodDescriptor.Marshaller<byte[]> {
+
+    public static ByteArrayMarshaller instance() {
+        return new ByteArrayMarshaller();
+    }
+
+    @Override
+    public InputStream stream(byte[] bytes) {
+        return new ByteArrayInputStream(bytes);
+    }
+
+    @Override
+    public byte[] parse(InputStream stream) {
+        byte[] buf = null;
+        try {
+            if (stream instanceof KnownLength) {
```

Comment:
Why do you check size less than 100? Is there a benefit that you don't want to run this code for greater lengths?